src/Data/Singletons/Internal.hs

```diff
@@ -97,6 +109,38 @@ class SingKind k where
 data SomeSing k where
   SomeSing :: Sing (a :: k) -> SomeSing k
 
+-- | An (explicitly bidirectional) pattern synonym for going between a
+-- term and the corresponding demoted term.
```

Comment:
I would phrase this as "... for going between a singleton and the corresponding demoted term."